webserver/views/datasets.py

```diff
@@ -308,7 +308,11 @@ def edit(dataset_id):
 def delete(dataset_id):
     ds = get_dataset(dataset_id)
     if ds["author"] != current_user.id:
-        raise Unauthorized("You can't delete this dataset.")
+        return jsonify(
```

Comment:
ah, right :+1:. Can we replace this with [`APIUnauthorized`](https://github.com/metabrainz/acousticbrainz-server/blob/0461cbe92aab16718703e25f5df589e4110895ca/webserver/views/api/exceptions.py#L18)?